io/src/pcd_grabber.cpp

```diff
@@ -387,7 +387,6 @@ pcl::PCDGrabberBase::PCDGrabberBase (const std::vector<std::string>& pcd_files,
 ///////////////////////////////////////////////////////////////////////////////////////////
 pcl::PCDGrabberBase::~PCDGrabberBase () throw ()
 {
-  stop ();
   delete impl_;
```

Comment:
Unfortunately it is not a good idea. The trigger function calls the outside class' (PCDGrabberBase) publish function. https://github.com/soyersoyer/pcl/blob/pcdgrabber_fix/io/src/pcd_grabber.cpp#L291
After a copy it will call the copied PCDGrabber's publish function, not the new.

I think the PCDGrabber is not a good "value class" candidate (it is polymorphic), so the copy constructor and assignment should be private/deleted. If the copy is necessary, it is achievable with a virtual clone function.